core/src/main/java/com/godaddy/commerce/core/jwt/CertBasedJwtCache.java

```diff
@@ -0,0 +1,117 @@
+package com.godaddy.commerce.core.jwt;
+
+import com.godaddy.commerce.core.errors.ProcessingException;
+import com.godaddy.commerce.core.Expirable;
+import com.godaddy.commerce.core.http.config.ConnectionProperty;
+import com.godaddy.commerce.core.http.config.EcommerceHost;
+import com.godaddy.commerce.core.http.HttpVerbWrapper;
+import io.vertx.ext.web.RoutingContext;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public enum CertBasedJwtCache {
+  INSTANCE;
+
+  private final static Logger LOGGER = LoggerFactory.getLogger(CertBasedJwtCache.class);
+  private final static Map<String, Expirable<JwtResponse>> CACHE = new HashMap<>();
+  private final static String SSO_TOKEN_HTTP_PATH = "/v1/secure/api/token";
+  private static String SSO_CONNECTION_PROPERTY_KEY = "SSO";
+
+  public static void setEnvType(String envType) {
+    // Use suffixed key for SSO in PCI/ICP.
+    SSO_CONNECTION_PROPERTY_KEY = envType == "PCI" || envType == "ICP"
+      ? String.format("SSO_%s", envType)
+      : "SSO";
```

Comment:
> [<img alt="szhang1" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.secureserver.net/szhang1) **Authored by [szhang1](https://github.secureserver.net/szhang1)**
_<time datetime="2020-11-16T18:02:32Z" title="Monday, November 16th 2020, 7:02:32 pm +01:00">Nov 16, 2020</time>_

---


`equals` should be used for string comparisons and Sonar will complain if you put the constant values on the right (due to potential `NullPointerException`s).

```suggestion
    SSO_CONNECTION_PROPERTY_KEY = "PCI".equals(envType) || "ICP".equals(envType)
      ? String.format("SSO_%s", envType)
      : "SSO";
```